reference/test/distributed/partition_kernels.cpp

```diff
@@ -0,0 +1,339 @@
+    assert_equal_data(partition->get_part_ids(),
+                      {2, 0, 1, 2, 0, 1, 0, 1, 2, 0});
+    assert_equal_data(partition->get_range_starting_indices(),
+                      {0, 0, 0, 2, 1, 2, 3, 3, 3, 4});
+    assert_equal_data(partition->get_part_sizes(), {5, 6, 5});
+}
+
+
+TYPED_TEST(Partition, BuildsFromMappingWithEmptyParts)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    gko::Array<comm_index_type> mapping{
+        this->ref, {3, 3, 0, 1, 1, 3, 0, 0, 1, 0, 1, 1, 1, 3, 3, 0}};
+    comm_index_type num_parts = 5;
+    gko::size_type num_ranges = 10;
+
+    auto partition = gko::distributed::Partition<
+        local_index_type, global_index_type>::build_from_mapping(this->ref,
+                                                                 mapping,
+                                                                 num_parts);
+
+    EXPECT_EQ(partition->get_size(), mapping.get_num_elems());
+    EXPECT_EQ(partition->get_num_ranges(), num_ranges);
+    EXPECT_EQ(partition->get_num_parts(), num_parts);
+    EXPECT_EQ(partition->get_num_empty_parts(), 2);
+    assert_equal_data(partition->get_range_bounds(),
+                      {0, 2, 3, 5, 6, 8, 9, 10, 13, 15, 16});
+    assert_equal_data(partition->get_part_ids(),
+                      {3, 0, 1, 3, 0, 1, 0, 1, 3, 0});
+    assert_equal_data(partition->get_range_starting_indices(),
+                      {0, 0, 0, 2, 1, 2, 3, 3, 3, 4});
+    assert_equal_data(partition->get_part_sizes(), {5, 6, 0, 5, 0});
+}
+
+
+TYPED_TEST(Partition, BuildsFromRanges)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    gko::Array<global_index_type> ranges{this->ref, {0, 5, 5, 7, 9, 10}};
+
+    auto partition = gko::distributed::Partition<
+        local_index_type, global_index_type>::build_from_contiguous(this->ref,
+                                                                    ranges);
+
+    EXPECT_EQ(partition->get_size(),
+              ranges.get_data()[ranges.get_num_elems() - 1]);
+    EXPECT_EQ(partition->get_num_ranges(), ranges.get_num_elems() - 1);
+    EXPECT_EQ(partition->get_num_parts(), ranges.get_num_elems() - 1);
+    EXPECT_EQ(partition->get_num_empty_parts(), 1);
+    assert_equal_data(partition->get_range_bounds(), {0, 5, 5, 7, 9, 10});
+    assert_equal_data(partition->get_part_ids(), {0, 1, 2, 3, 4});
+    assert_equal_data(partition->get_range_starting_indices(), {0, 0, 0, 0, 0});
+    assert_equal_data(partition->get_part_sizes(), {5, 0, 2, 2, 1});
+}
+
+
+TYPED_TEST(Partition, BuildsFromRangeWithSingleElement)
+{
+    using local_index_type = typename TestFixture::local_index_type;
+    using global_index_type = typename TestFixture::global_index_type;
+    gko::Array<global_index_type> ranges{this->ref, {0}};
```

Comment:
what happen when it is not zero?

Reply:
I've clarified the documentation, that the first element of the ranges needs always to be 0. Still, it wouldn't matter, you would still get an empty partition, but with `range_bounds = {0}`, regardless of what element was passed.

Reply:
Thanks for clarification